drivers/interrupt_controller/intc_shared_irq.c

```diff
@@ -125,9 +125,16 @@ int shared_irq_initialize(const struct device *dev)
 	return 0;
 }
 
+#define F1(x) 1
+#define INST_SUPPORTS_DEP_ORDS_CNT(index)  \
+	(FOR_EACH(F1, (+), DT_INST_SUPPORTS_DEP_ORDS(index)) - 1)
+
 #define SHARED_IRQ_CONFIG_FUNC(index)					\
 void shared_irq_config_func_##index(void)				\
 {									\
+	__ASSERT(INST_SUPPORTS_DEP_ORDS_CNT(index) <=			\
+		 CONFIG_SHARED_IRQ_NUM_CLIENTS,				\
+		 "Instance can't have more clients than configured");	\
```

Comment:
Is that added on purpose ?
You're removing it in the next commit.

Reply:
Yes I added it on purpose as in this commit the array size is still determined with kconfig.
But it is really superfluous so I'll remove it.